srcpkgs/efibootmgr/files/kernel.d/efibootmgr.post-install

```diff
@@ -28,18 +28,26 @@ major_version=$(echo $PKGNAME | cut -c 6-)
 # look for previous entry for this major kernel version
 existing_entry=$(efibootmgr | grep "Void Linux with kernel ${major_version}")
 
-# get the boot order
-# this is required because when in the next step the existing entry is removed,
-# it is also removed from the order so it needs to be restored later
-bootorder=$(efibootmgr |grep "BootOrder: " |cut -c 12-)
-
 # if existing, remove it
 if [ "$existing_entry" != "" ]; then
-  /etc/kernel.d/post-remove/50-efibootmgr $PKGNAME
+	/etc/kernel.d/post-remove/50-efibootmgr $PKGNAME
 fi
 
-# create the new entry
-efibootmgr -qc $args -L "Void Linux with kernel ${major_version}" -l /vmlinuz-${VERSION} -u "${OPTIONS}"
-
-# restore the boot order
-efibootmgr -qo $bootorder
+# if using strict efi suffix
+if [ "x${USE_STRICT_EFI_SUFFIX}" = x1 ]; then
+	# rename vmlinuz
+	if [ -f /boot/vmlinuz-${VERSION} ]; then
+		cp -f /boot/vmlinuz-${VERSION} /boot/vmlinuz-${VERSION}.efi
+	fi
+	# create the new entry
+	efibootmgr -qc $args -L "Void Linux with kernel ${major_version}" -l /vmlinuz-${VERSION}.efi -u "${OPTIONS}"
+else
+	# remove efi
+	if [ -f /boot/vmlinuz-${VERSION}.efi ]; then
+		# since we copy vmlinuz-... to vmlinuz-....efi, we don't need
+		# to copy it back.
+		rm -f /boot/vmlinuz-${VERSION}.efi
+	fi
+	# create the new entry
+	efibootmgr -qc $args -L "Void Linux with kernel ${major_version}" -l /vmlinuz-${VERSION} -u "${OPTIONS}"
```

Comment:
Repeating the `efibootmgr` command in both the `if` and `else` blocks just invites mistakes if people want to modify loader behavior in the future. Better to define a variable, `kernel_path` or something, and set
```sh
kernel_path="/vmlinuz-${VERSION}"
```
here and
```sh
kernel_path="[...]/vmlinuz-${VERSION}.efi"
```
in the above `if` block. (See my general comment for how I think `[...]` should be handled.) This way, you can make one call to
```sh
efibootmgr -qc $args -L "Void Linux with kernel ${major_version}" -l "${kernel_path}" -u "${OPTIONS}"
```
after the block.